src/components/Common/MentionDropdown.tsx

```diff
@@ -0,0 +1,166 @@
+import { t } from "i18next";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+
+import { Avatar } from "@/components/Common/Avatar";
+
+import useSlug from "@/hooks/useSlug";
+
+import routes from "@/Utils/request/api";
+import useQuery from "@/Utils/request/useQuery";
+import { formatDisplayName } from "@/Utils/utils";
+
+interface MentionsDropdownProps {
+  onSelect: (user: { id: string; username: string }) => void;
+  position: { top: number; left: number };
+  filter: string;
+  containerRef: React.RefObject<HTMLTextAreaElement>;
+  parentRef?: React.RefObject<HTMLElement>;
+}
+
+const KEYS = {
+  ENTER: "Enter",
+  ARROW_UP: "ArrowUp",
+  ARROW_DOWN: "ArrowDown",
+  ESCAPE: "Escape",
+} as const;
+
+const MentionsDropdown: React.FC<MentionsDropdownProps> = ({
+  onSelect,
+  position,
+  filter,
+  containerRef,
+  parentRef,
+}) => {
+  const facilityId = useSlug("facility");
+  const { data, loading } = useQuery(routes.getFacilityUsers, {
+    pathParams: { facility_id: facilityId },
+  });
```

Comment:
Although works, prefer using tanstack's useQuery instead (especially in newer places). this is deprecated. Docs: https://docs.ohc.network/docs/care/development/data-fetching-in-care/

Reply:
got it changed all